.github/workflows/CI.yml

```
@@ -0,0 +1,69 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
```

We might want to specify a particular version of python - we seem to be using `3.11` a lot: 
```suggestion
      - uses: actions/setup-python@v5
        with:
          python-version: '3.11'
          cache: pip
```